tasks/puppet/src/main/java/com/walmartlabs/concord/plugins/puppet/PuppetClient.java

```diff
@@ -215,48 +227,31 @@ private static <T> T withRetry(int retryCount, long retryInterval, Callable<T> c
      *
      * @param cfg config with certificate settings
      * @return http client with certificate settings from the config
-     * @throws Exception
      */
-    private OkHttpClient createClient(PuppetConfiguration cfg) throws Exception {
-        OkHttpClient.Builder clientBuilder = new OkHttpClient.Builder()
-                .connectTimeout(cfg.getConnectTimeout(), TimeUnit.SECONDS)
-                .readTimeout(cfg.getReadTimeout(), TimeUnit.SECONDS)
-                .writeTimeout(cfg.getWriteTimeout(), TimeUnit.SECONDS);
+    private HttpClient createClient(PuppetConfiguration cfg) throws Exception {
+        var clientBuilder = HttpClient.newBuilder()
+                .connectTimeout(Duration.ofSeconds(cfg.getConnectTimeout()));
+
+        if (cfg.getHttpVersion() == PuppetConfiguration.HttpVersion.HTTP_1_1) {
+            clientBuilder.version(HttpClient.Version.HTTP_1_1);
+        } else if (cfg.getHttpVersion() == PuppetConfiguration.HttpVersion.HTTP_2) {
+            clientBuilder.version(HttpClient.Version.HTTP_2);
+        }
```

Comment:
Throw an error for unsupported versions?

Reply:
I was trying to avoid the unlikely-but-not-impossible situation of an existing workflow having a `httpVersion` variable that would sneak in here in runtime-v1.

Really the only reason this is here is for a just-in-case situation and because there's something off with wiremock and `HttpClient` where it allows it to start an HTTP/2 connection but results in a reset connection. So forcing it down to 1.1 in the unit tests is necessary.